Add online count to Lobby

GrantBartlett opened the pull request:
![online-count](https://user-images.githubusercontent.com/6104940/40586626-9876949a-61bc-11e8-9118-49c304dc2902.png)

Could also possibly look at: 
- Changing the online count when the channel is changed.
- Making the Text customizable in an INI file somewhere.

Rampastring commented on DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:
```diff
@@ -239,6 +247,10 @@ public override void Initialize()
+            AddChild(lblOnline);
+
+            CnCNetPlayerCountTask.CnCNetGameCountUpdated += OnCnCNetGameCountUpdated;
+            CnCNetPlayerCountTask.ForceRefresh();
```
Why is ForceRefresh necessary?

Rampastring commented on DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:
```diff
@@ -247,6 +259,14 @@ public override void Initialize()
+            lock (locker)
+            {
+                lblOnline.Text = "Online: " + e.PlayerCount.ToString();
+            }
```
Locking might not be necessary here, although MainMenu does it too. I'll check later if it's really needed.

Rampastring commented on DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:
```diff
@@ -224,6 +226,12 @@ public override void Initialize()
+            lblOnline.Name = "lblOnline";
+            lblOnline.ClientRectangle = new Rectangle(310, 14, 0, 0);
+            lblOnline.FontIndex = 1;
+            lblOnline.Text = "";
```
Call lblOnline.Disable() here to make it disabled by default. Then you can enable it for YR in CnCNetLobby.ini.

GrantBartlett commented:
- Removed locker
- Replaced ForceUpdater with a cached online count.
- Seperated lblOnline and lblOnlineCount into two separate labels so they can be edited fully.
- Lobby count disabled by default

GrantBartlett commented:
@Rampastring ^

Rampastring commented on DXMainClient/Domain/Multiplayer/CnCNet/CnCNetPlayerCountTask.cs:
```diff
@@ -86,6 +82,8 @@ private static int GetCnCNetPlayerCount()
             }
         }
+
+        public static int PlayerCount { get; set; }
```
Move this to the beginning of the class (like the other fields) and make the setter private.
